src/System.Private.CoreLib/shared/System/String.Comparison.cs

```diff
@@ -251,14 +251,10 @@ public static int Compare(string? strA, string? strB, StringComparison compariso
         // to determine whether it is lexicographically less, equal, or greater, and then a
         // negative integer, 0, or a positive integer is returned; respectively.
         //
-        public static int Compare(string? strA, string? strB, CultureInfo culture, CompareOptions options)
+        public static int Compare(string? strA, string? strB, CultureInfo? culture, CompareOptions options)
         {
-            if (culture == null)
-            {
-                throw new ArgumentNullException(nameof(culture));
-            }
-
-            return culture.CompareInfo.Compare(strA, strB, options);
+            CultureInfo compareCulture = culture ?? CultureInfo.CurrentCulture;
```

Comment:
Do we need a test for this?

Reply:
Is this a change you and @safern agreed upon?

We'd discussed this in the meeting, and I thought the verdict was to leave it as non-null / throwing for null; this is different from APIs that accept IFormatProvider, where we do accept null... we have very few public APIs that directly take a CultureInfo, and there isn't really similar precedent there for allowing null to be passed.

If I'm remembering incorrectly and the group did decide to allow null here, I'm ok with that.  But yeah, I would have expected this change to cause a test failure; that it doesn't suggests we have a test gap.  We should be either fixing or adding tests for this.

Reply:
I think the group decided to accept null here from what I can remember. Actually I did this change by following @terrajobst's notes: https://github.com/dotnet/apireviews/pull/97/files#diff-3d5df95ecd3f337acffecda634fc9ba8R32

cc: @tannergooding @tarekgh @bartonjs were on that meeting so they might have other insight.

Reply:
There are videos, right?

Reply:
Right. I'll dig them up and see what was the final consensus.

Reply:
After looking at the video it seems like we agreed on being consistent on all APIs that took a CultureInfo to accept null and default it to CurrentCulture.

Reply:
> We should be either fixing or adding tests for this.

I'll follow up on the tests.